compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDistributeCoresAndObjectFifos.cpp

```diff
@@ -244,17 +250,23 @@ class AMDAIEUnrollLocalLoops : public OpRewritePattern<scf::ForOp> {
       for (auto it = loopBodyBlock->begin(); it != std::next(srcBlockEnd);
            it++) {
         if (auto dmaOp = dyn_cast<AMDAIE::DmaCpyNdOp>(*it)) {
-          uint8_t sourceMemSpaceInt = dmaOp.getSourceMemorySpaceAsUInt();
-          uint8_t targetMemSpaceInt = dmaOp.getTargetMemorySpaceAsUInt();
-          if (targetMemSpaceInt > sourceMemSpaceInt) {
+          std::optional<uint8_t> sourceMemSpaceInt =
+              dmaOp.getSourceMemorySpaceAsUInt();
+          std::optional<uint8_t> targetMemSpaceInt =
+              dmaOp.getTargetMemorySpaceAsUInt();
+          if (!sourceMemSpaceInt || !targetMemSpaceInt) {
+            return dmaOp.emitOpError()
```

Comment:
Emiting error from pattern. Assert, or notifyMatchFailure?

Reply:
Updated to `notifyMatchFailure`